ci/tests/general/file_reader.py

```diff
@@ -0,0 +1,448 @@
+# Copyright (C) 2016 iNuron NV
+#
+# This file is part of Open vStorage Open Source Edition (OSE),
+# as available from
+#
+#      http://www.openvstorage.org and
+#      http://www.openvstorage.com.
+#
+# This file is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License v3 (GNU AGPLv3)
+# as published by the Free Software Foundation, in version 3 as it comes
+# in the LICENSE.txt file of the Open vStorage OSE distribution.
+#
+# Open vStorage is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY of any kind.
+import os
+import re
+import time
+import inspect
+
+from shutil import copyfile
+from stat import *
+from datetime import date, datetime, timedelta
+
+
+class LogFileTimeParser(object):
+    """
+    # @TODO Support for rotated log files - currently using the current year for 'Jan 01' dates.
+    Extracts parts of a log file based on a start and end date
+    Uses binary search logic to speed up searching
+    Can filter out errors based on string patterns
+
+    Hardly uses any memory - cpu intensive though
+    Writes results to a file and returns the contents of this file to save memory
+
+    Common usage: validate log files during testing
+    """
+    version = "0.01b"
+
+    # Set some initial values
+    BUF_SIZE = 4096  # handle long lines, but put a limit to them
+    REWIND = 100  # arbitrary, the optimal value is highly dependent on the structure of the file
+    LIMIT = 75  # arbitrary, allow for a VERY large file, but stop it if it runs away
+    PATHS_TO_FILES = ['/var/log/upstart/ovs-workers.log', '/var/log/upstart/ovs-webapp-api.log', '/var/log/ovs/lib.log',
```

Comment:
perhaps we can also look into alba proxy / voldrv / asds / etcd for errors as a next step